### PR TITLE
Add libdef for react-i18next v6.x.x for flow v0.53.x- (#1279)

### PR DESCRIPTION
**Pull request by TLadd**
Copied over the v2.x.x libdef, added the additional exports that have been added to match the latest release of react-i18next, and made modifications to be 0.53.x- compatible.

### CODE REVIEW

File: definitions/npm/react-i18next_v6.x.x/test_react-i18next_v6.x.x.js [hunk 1]
```diff
@@ -0,0 +1,175 @@
...
+class FlowErrorClassComp extends React.Component<Props> {
+  render() {
+    // $ExpectError - wrong argument type
+    const { s, t } = this.props;
+    return <div prop={t({})} />; // misuse of t()
+  }
+}
+
+// $ExpectError - too few arguments
+translate();
```
Comment (stunaz) [inline definitions/npm/react-i18next_v6.x.x/test_react-i18next_v6.x.x.js]: why is that an error, we can have `translate()` which means we are using defaultNS ==> https://react.i18next.com/components/translate-hoc.html

File: definitions/npm/react-i18next_v6.x.x/flow_v0.53.x-/react-i18next_v6.x.x.js [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+    nsMode: "default" | "fallback",
+    bindi18n: false | string,
+    bindStore: false | string,
+    withRef: boolean,
+    translateFuncName: string,
+    i18n: Object
+  }>;
+
+  declare function translate<OP, P>(
+    locales: Locales,
```
Comment (stunaz) [inline definitions/npm/react-i18next_v6.x.x/flow_v0.53.x-/react-i18next_v6.x.x.js]: locales is an optional argument too ==> https://react.i18next.com/components/translate-hoc.html

**Comment by stunaz** [comment]
@TLadd I found an issue with the flow-typed , `translate` can have no parameters which means it is using defaultNS , have a look at https://react.i18next.com/components/translate-hoc.html or better at the source code https://github.com/i18next/react-i18next/blob/master/src/translate.js#L20

**Comment by TLadd** [comment]
You are correct. Would you like to submit a PR correcting the issue?

**Comment by stunaz** [comment]
will do :-) will ping you to review

**Pull request merged**